src/main/scala/vsys/blockchain/contract/DataEntry.scala

```diff
@@ -83,16 +56,17 @@ object DataEntry {
     bytes.headOption match {
       case Some(b) =>
         DataType.fromByte(b) match {
-          case Some(dt: DataType.DataTypeVal) if (dt.lenFixed && checkDataType(bytes.drop(1).take(dt.maxLen), dt)) =>
+          case Some(dt: DataType.DataTypeVal[_]) if (dt.lenFixed && dt.validator(bytes.drop(1).take(dt.maxLen))) =>
             Right((DataEntry(bytes.drop(1).take(dt.maxLen), dt), bytes.drop(1 + dt.maxLen)))
-          case Some(dt: DataType.DataTypeVal) if (!dt.lenFixed && dt.maxLen <= Short.MaxValue && checkDataType(bytes.drop(1).take(2 + Shorts.fromByteArray(bytes.drop(1).take(2))), dt)) =>
-            Right((DataEntry(bytes.drop(1).take(2 + Shorts.fromByteArray(bytes.drop(1).take(2))), dt), bytes.drop(3 + Shorts.fromByteArray(bytes.drop(1).take(2)))))
+          case Some(dt: DataType.DataTypeVal[_]) if (!dt.lenFixed && dt.maxLen <= Short.MaxValue
+            && dt.validator(bytes.drop(1).take(2 + Shorts.fromByteArray(bytes.drop(1).take(2))))) =>
```

Comment:
I am worried about the `Shorts.fromByteArray` here, if `take(2)` return an empty `Array[Byte]` what will happen here? or we need an extra condition `bytes.length >=3`.